### PR TITLE
[libspatialindex] Update to 1.9.3

### PR DESCRIPTION
**Pull request by longhuan2018**
update libspatialindex to 1.9.3

### CODE REVIEW

File: ports/libspatialindex/portfile.cmake [hunk 1]
```diff
@@ -1,16 +1,17 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO libspatialindex/libspatialindex
-    REF 1.9.0
-    SHA512   368537e9bfe52db96486a1febfabe035f9f7714fd1cb50450e3ab89d51c5ffffb0e2ea219e08bee34f772ba9813a3a7f9e63d8b8946887ce83811ef68d17d1cc
+    REF 1.9.3
+    SHA512   d4c608abbd631dc163b7b4fb6bf09dee3e85ce692a5f8875d51f05a26e09c75cd17dff1ed9d2c232a071f0f5864d21d877b4cbc252f3416896db24dfa3fa18cb
```
Comment (NancyLi1013) [inline ports/libspatialindex/portfile.cmake:5]:
```suggestion
    SHA512 d4c608abbd631dc163b7b4fb6bf09dee3e85ce692a5f8875d51f05a26e09c75cd17dff1ed9d2c232a071f0f5864d21d877b4cbc252f3416896db24dfa3fa18cb
```

File: ports/libspatialindex/portfile.cmake [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
-	PATCHES
+    PATCHES
         static.patch
 )
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
-	OPTIONS -DCMAKE_DEBUG_POSTFIX=d -DSIDX_BUILD_TESTS:BOOL=OFF
+    PREFER_NINJA
+    OPTIONS -DCMAKE_DEBUG_POSTFIX=d -DSIDX_BUILD_TESTS:BOOL=OFF
```
Comment (NancyLi1013) [inline ports/libspatialindex/portfile.cmake:14]:
```suggestion
    OPTIONS 
        -DCMAKE_DEBUG_POSTFIX=d 
        -DSIDX_BUILD_TESTS:BOOL=OFF
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Failed on x64-windows-static and x64-windows-static-md:

```
DLLs should not be present in a static build, but the following DLLs were found:

    D:/packages/libspatialindex_x64-windows-static/bin/spatialindex-64.dll
    D:/packages/libspatialindex_x64-windows-static/bin/spatialindex_c-64.dll
    D:/packages/libspatialindex_x64-windows-static/debug/bin/spatialindex-64.dll
    D:/packages/libspatialindex_x64-windows-static/debug/bin/spatialindex_c-64.dll

There should be no bin\ directory in a static build, but D:\packages\libspatialindex_x64-windows-static\bin is present.
There should be no debug\bin\ directory in a static build, but D:\packages\libspatialindex_x64-windows-static\debug\bin is present.
If the creation of bin\ and/or debug\bin\ cannot be disabled, use this in the portfile to remove them

    if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
        file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
    endif()

Found 2 error(s). Please correct the portfile:
```

File: ports/libspatialindex/portfile.cmake [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
-	PATCHES
+    PATCHES
         static.patch
 )
 
+if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
+    set(BUILD_SHARED_LIBS ON)
+else()
+    set(BUILD_SHARED_LIBS OFF)
+endif()
```
Comment (NancyLi1013) [inline ports/libspatialindex/portfile.cmake:15]:
```suggestion
```
By default, vcpkg_configure_cmake() will pass in the appropriate setting for BUILD_SHARED_LIBS. So there is no need to add these lines.
Please see https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#choose-either-static-or-shared-binaries

File: ports/libspatialindex/portfile.cmake [hunk 4]
```diff
@@ -1,16 +1,25 @@
...
+
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
-	OPTIONS -DCMAKE_DEBUG_POSTFIX=d -DSIDX_BUILD_TESTS:BOOL=OFF
+    OPTIONS
+        -DCMAKE_DEBUG_POSTFIX=d
+        -DSIDX_BUILD_TESTS:BOOL=OFF
+        -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}
```
Comment (NancyLi1013) [inline ports/libspatialindex/portfile.cmake:22]:
```suggestion
```
The same as above.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

File: ports/libspatialindex/portfile.cmake [hunk 5]
```diff
@@ -1,16 +1,19 @@
...
-    SHA512   368537e9bfe52db96486a1febfabe035f9f7714fd1cb50450e3ab89d51c5ffffb0e2ea219e08bee34f772ba9813a3a7f9e63d8b8946887ce83811ef68d17d1cc
+    REF 1.9.3
+    SHA512 d4c608abbd631dc163b7b4fb6bf09dee3e85ce692a5f8875d51f05a26e09c75cd17dff1ed9d2c232a071f0f5864d21d877b4cbc252f3416896db24dfa3fa18cb
     HEAD_REF master
-	PATCHES
+    PATCHES
         static.patch
 )
 
+
```
Comment (NancyLi1013) [inline ports/libspatialindex/portfile.cmake:11]:
```suggestion
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Please run `vcpkg x-add-version libspatialindex` to solve the failures on x86-windows.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your update @longhuan2018.

**Comment by m-kuhn** [comment]
@longhuan2018 I created a PR upstream in anticipation of an easier upgrade path in the future

**Comment by strega-nil-ms** [comment]
LGTM, thanks @longhuan2018 :)

**Pull request merged**